### PR TITLE
Use derive_new and getset macros for amethyst_ui

### PR DESCRIPTION
**Pull request by Areredify**
## Description

As per issue ([#1838]), new is replaced with macros, getset is exported for future use.

Please inspect changes in text.rs and selection.rs more closely, I'm not completely sure in those.

## PR Checklist

By placing an x in the boxes I certify that I have:

- [ ] Updated the content of the book if this PR would make the book outdated.
- [ ] Added a changelog entry if this will impact users, or modified more than 5 lines of Rust that wasn't a doc comment.
- [ ] Added unit tests for new code added in this PR.
- [x] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.

If this modified or created any rs files:

- [x] Ran `cargo +stable fmt --all`
- [x] Ran `cargo clippy --all --features "empty"`
- [x] Ran `cargo test --all --features "empty"`

### CODE REVIEW

File: amethyst_ui/src/sound.rs [hunk 1]
```diff
@@ -89,19 +89,14 @@ impl<'a, 'b> SystemDesc<'a, 'b, UiSoundSystem> for UiSoundSystemDesc {
 /// Handles any dispatches `UiPlaySoundAction`s and plays the received
 /// sounds through the set `Output`.
 #[derive(Debug)]
+/// Constructs a default `UiSoundSystem`. Since the `event_reader`
+/// will automatically be fetched when the system is set up, this should
+/// always be used to construct the `UiSoundSystem`.
+#[derive(new)]
```
Comment (jojolepro) [inline amethyst_ui/src/sound.rs]: same here, does this doc generates properly?

File: amethyst_ui/src/event_retrigger.rs [hunk 2]
```diff
@@ -70,22 +70,14 @@ where
 /// to incoming events and calling `apply` on the respective `Retrigger`
 /// components.
 #[derive(Debug)]
+/// Constructs a default `EventRetriggerSystem`. Since the `event_reader`
+/// will automatically be fetched when the system is set up, this should
+/// always be used to construct `EventRetriggerSystem`s.
+#[derive(new)]
```
[the remark this answers is not in the document]
Comment (Areredify) [reply amethyst_ui/src/event_retrigger.rs]: It does not, should I change it back to explicit new?

File: amethyst_ui/src/selection.rs [hunk 3]
```diff
@@ -78,23 +78,13 @@ where
...
 {
-    /// Creates a new `SelectionKeyboardSystem`.
-    pub fn new(window_reader_id: ReaderId<Event>) -> Self {
-        Self {
-            window_reader_id,
-            phantom: PhantomData,
-        }
-    }
+    window_reader_id: ReaderId<Event>,
+    phantom: PhantomData<G>,
```
[the remark this answers is not in the document]
Comment (Areredify) [reply amethyst_ui/src/selection.rs]: PhantomData is generated automaticly

File: amethyst_ui/src/selection.rs [hunk 4]
```diff
@@ -78,23 +78,13 @@ where
...
-pub struct SelectionKeyboardSystem<G> {
-    window_reader_id: ReaderId<Event>,
-    phantom: PhantomData<G>,
-}
-
-impl<G> SelectionKeyboardSystem<G>
+#[derive(Debug, new)]
+pub struct SelectionKeyboardSystem<G>
 where
     G: Send + Sync + 'static + PartialEq,
```
[the remark this answers is not in the document]
Comment (Areredify) [reply amethyst_ui/src/selection.rs]: New used to have a type bound, is it not important?

**Pull request closed without merging**